api/src/main/java/org/jboss/marshalling/util/IdentityIntMap.java

```diff
@@ -24,11 +24,17 @@
  * An efficient identity object map whose keys are objects and whose values are {@code int}s.
  */
 public final class IdentityIntMap<T> implements Cloneable {
+
+    private static int REINITIALIZE_CAPACITY_THRESHOLD = Integer.getInteger("jboss.marshalling.identityint.reinitialize.capacity.threshold", 1 << 16); // 65536
```

Comment:
This should be more like: 
```suggestion
    private static int REINITIALIZE_CAPACITY_THRESHOLD = Integer.parseInt(AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getProperty("jboss.marshalling.identityint.reinitialize.capacity.threshold", Integer.toString(1 << 16)));
```
I did not check this for correctness but the important point is to get the property from a privileged block.

Reply:
Thanks for the review. I've updated the patch by adding a static initializer that uses AccessController.doPrivileged() if the security manager is enabled.